### ISSUE TITLE
[Discussion] Support Multiple Backends

### ISSUE THREAD

**Issue by sxjscience** [dmlc/gluon-nlp#1330]
Many DL frameworks are moving to use numpy as their default way for writing DL models. Thus, we may try to support multiple-backends for the model-zoo in GluonNLP. As long as the backend supports numpy-like interface, e.g., [JAX](https://github.com/google/jax) , [tf.experimental.numpy](https://www.tensorflow.org/api_docs/python/tf/experimental/numpy), and the ongoing effort in PyTorch: https://github.com/pytorch/pytorch/issues/38349, we are able to reuse most of the [existing model implementations](https://github.com/dmlc/gluon-nlp/tree/master/src/gluonnlp/models) and make it work for different DL frameworks. One advantage is that the user is able to try out TF-2.0, Jax and PyTorch very easily.

What do you think? @dmlc/gluon-nlp-committers

**Comment by leezu** [comment]
We can extend the Gluon interface/create a independent modeling library to enable switching backends

**Comment by sxjscience** [comment]
I think after some rounds of discussion, the plan will be to add an additional PyTorch-based backend in GluonNLP.

**Comment by sxjscience** [comment]
Why close this?

**Comment by szha** [comment]
As you stated above that the discussion has concluded that we will support pytorch, I thought we can close the discussion thread and start listing the tasks.

**Comment by szha** [comment]
Feel free to reopen if you think this decision needs to be revisited with more discussion.
